nflow-rest-api-common/src/main/java/io/nflow/rest/v1/ResourceBase.java

```diff
@@ -158,35 +157,42 @@ public boolean updateWorkflowInstance(final long id,
         .addStatuses(statuses.toArray(new WorkflowInstanceStatus[statuses.size()]))
         .setBusinessKey(businessKey)
         .setExternalId(externalId)
-        .setIncludeCurrentStateVariables(includeStrings.contains(currentStateVariables))
-        .setIncludeActions(includeStrings.contains(actions))
-        .setIncludeActionStateVariables(includeStrings.contains(actionStateVariables))
+        .setIncludeCurrentStateVariables(propertyIncludes.contains(currentStateVariables))
+        .setIncludeActions(propertyIncludes.contains(actions))
+        .setIncludeActionStateVariables(propertyIncludes.contains(actionStateVariables))
         .setMaxResults(maxResults)
         .setMaxActions(maxActions)
         .setQueryArchive(queryArchive)
-        .setIncludeChildWorkflows(includeStrings.contains(childWorkflows))
+        .setIncludeChildWorkflows(propertyIncludes.contains(childWorkflows))
         .setStateVariable(stateVariableKey, stateVariableValue)
         .build();
     Stream<WorkflowInstance> instances = workflowInstances.listWorkflowInstancesAsStream(q);
-    Set<WorkflowInstanceInclude> parseIncludeEnums = parseIncludeEnums(include);
-    return instances.map(instance -> listWorkflowConverter.convert(instance, parseIncludeEnums, queryArchive));
-  }
-
-  private Set<WorkflowInstanceInclude> parseIncludeEnums(String include) {
-    return parseIncludeStrings(include).map(INCLUDE_STRING_TO_ENUM::get).filter(Objects::nonNull)
-        .collect(toCollection(HashSet::new));
+    return instances.map(instance -> listWorkflowConverter.convert(instance, propertyIncludes, queryArchive));
   }
 
-  private Stream<String> parseIncludeStrings(String include) {
-    return Stream.of(trimToEmpty(include).split(","));
+  private Set<ApiWorkflowInstanceInclude> resolveIncludes(Set<ApiWorkflowInstanceInclude> includes, String include) {
+    Set<ApiWorkflowInstanceInclude> allIncludes = new HashSet<>();
+    if (includes != null) {
+      allIncludes.addAll(includes);
+    }
+    if (isNotBlank(include)) {
+      allIncludes.addAll(stream(include.split(include))
```

Comment:
No reason to first collect a set and then add to another. PS the split was buggy. I guess you did not leave any tests for it?

How about:
```java
stream(include.split(","))
          .map(ApiWorkflowInstanceInclude::fromValue)
          .filter(Objects::nonNull)
          .forEach(allIncludes::add);
```

Reply:
fixed